### ISSUE TITLE
Vulkan support

### ISSUE THREAD

**Issue by NULUSIOS** [mamedev/mame#754]
I submit this as a feature request, but is more to get feedback on any possible discussion and progress on the subject.

Vulkan seems to be big for most platforms and will allow for easier mobile presence.
Major gfx hardware now has Vulkan support.

**Comment by galibert** [comment]
There's something to discuss?  Through the bgfx backend, mame already
supports gl, gles, metal and dx11.  That's all of what's actually out
there.  Soon bgfx will also support vulkan and dx12 (which are very close),
so we'll support them too.  That's it.

**Comment by NULUSIOS** [comment]
Ah sorry didn't realize bgfx actually hides the whole rendering subsystem behind it, thought it was limited to effects and could be disabled.

**Comment by rb6502** [comment]
Specifically, the non-BGFX rendering paths are becoming deprecated.  We plan to run things like GPU computing and hardware rasterization of 3D games through BGFX since it abstracts everything for all the platforms we run on.

**Comment by Luke-Nukem** [comment]
A quick question about this. If Vulkan is implemented, how hard/easy would it be to then get a shader system much like the current HLSL system set up, using Vulkan's SPIR-V shader language? (Including for Vectors).

My only experience with MAME so far is reimplementing some old patches for Vectors on a slightly older version of MAME; so my knowledge is limited.

**Comment by rb6502** [comment]
Again, BGFX abstracts everything.  Once Vulkan is fully implemented, @MooglyGuy 's existing BGFX shader chains will Just Work.

**Comment by Luke-Nukem** [comment]
Ah, I think I misunderstood what BGFX actually was.
I do have another question though, how hard would it be to use BGFX for vector games? I haven't had much time to look further in to it due to study, but it's something I could be interested in trying if I can get a few pointers on where to start and what is relevant.

**Comment by belegdol** [comment]
Initial Vulkan support landed in 0.215

**Comment by cuavas** [comment]
I wouldn't consider this done yet - there are multiple serious issues.

**Comment by angelosa** [comment]
I personally use vulkan as my BGFX daily driver, is this still worth being open?

**Comment by cuavas** [comment]
Yeah, the 2023 update to BGFX seems to have usable Vulkan support.
